test/test_dynamic_shapes.py

```diff
@@ -423,6 +423,47 @@ def test_sym_ceil(self):
         self.assertIsInstance(r, torch.SymInt, msg=type(r))
         self.assertExpectedInline(str(shape_env.guards[1][0]), """Eq(3*s0, 15)""")
 
+    def test_sym_ite(self):
+        shape_env = ShapeEnv()
+        t = create_symint(shape_env, 5)
+        f = create_symint(shape_env, 4)
+        b1 = True
+        r1 = torch.fx.experimental.symbolic_shapes.sym_ite(b1, t, f)
+        self.assertTrue(r1 is t)
+        b2 = False
+        r2 = torch.fx.experimental.symbolic_shapes.sym_ite(b2, t, f)
+        self.assertTrue(r2 is f)
+        b3 = t == 5
+        r3 = torch.fx.experimental.symbolic_shapes.sym_ite(b3, t, f)
+        self.assertEqual(r3, 5)
```

Comment:
This test phrasing is a bit dorky because this is generating the guard. You want some test before this assertEqual showing that there is no guard created just because you sym_ite an expression.